internal/lm/nvml.go

```diff
@@ -193,3 +201,59 @@ func isMPSCapable(manager resource.Manager) (bool, error) {
 	}
 	return true, nil
 }
+
+func newGPUModeLabeler(manager resource.Manager) (Labeler, error) {
+	devices, err := manager.GetDevices()
+	if err != nil {
+		return nil, err
+	}
+	if len(devices) == 0 {
+		// no devices, return empty labels
+		return empty{}, nil
```

Comment:
Note that if `devices` is empty we return no labels, but if `classes` is empty, we return  `unknown` as the label. Which is the desired behaviour?

Reply:
They both are different behaviours. 

Devices being empty specifies no devices are present on the node so no need to label the node.

Classes are not being present (which is highly unlikely) is a state that devices are there on the node but we are not able to determine their mode. So, the label is `unknown`